### PR TITLE
Added basic simulation capabilities to IPv8

### PR DESCRIPTION
**Pull request by devos50**
This PR:

 - Adds basic simulation functionality to IPv8, including documentation and a code snippet on how to use this functionality.

### CODE REVIEW

**Comment by devos50** [comment]
validate

File: simulation/discrete_loop.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import asyncio
+import heapq
+
+
+class DiscreteLoop(asyncio.AbstractEventLoop):
+    """
+    A discrete asyncio loop, using async/await.
```
Comment (qstokkink) [inline simulation/discrete_loop.py:7]: How is that different from the normal event loop?

File: simulation/discrete_loop.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    This loop can, for example, be used during simulations with IPv8.
```
Comment (qstokkink) [inline simulation/discrete_loop.py:8]: Only for example? What else.

Also, how do I use this class?

**Comment by devos50** [comment]
validate

**Review by qstokkink — APPROVED** [review]

**Pull request merged**